lib/internal/url.js

```diff
@@ -1377,8 +1377,11 @@ function getPathFromURL(path) {
   return isWindows ? getPathFromURLWin32(path) : getPathFromURLPosix(path);
 }
 
+const percentRegEx = /%/g;
 function getURLFromFilePath(filepath) {
   const tmp = new URL('file://');
+  if (filepath.indexOf('%') !== -1)
```

Comment:
Two questions.
1. Why not `filepath.includes('%')`?
2. Why the `if` in the first place? IMHO this is not a such a hot path that it needs this micro-optimization.

Reply:
indexOf check is a performance optimization yes.  This is used for every module resolved, so yes trying to optimize for hot path (thinking of catering to 1000s of resolutions).